### PR TITLE
Unique counts, ISSN in title entries. daterange for usage

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 1]
```diff
@@ -245,37 +245,27 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
   }
 
   Future<UUID> upsertTeEntry(TenantPgPool pool, SqlConnection con, RoutingContext ctx,
-                             String counterReportTitle, String match) {
+                             String counterReportTitle, String printIssn, String onlineIssn) {
```
Comment (julianladisch) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:248]: Declaring `final String onlineIssn` avoids the additional `match` variable.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 2]
```diff
@@ -245,37 +245,27 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
...
-                  + " kbManualMatch)"
-                  + " VALUES ($1, $2, $3, $4, $5, $6)"
-                  + " ON CONFLICT (counterReportTitle) DO NOTHING")
-                  .execute(Tuple.tuple(List.of(UUID.randomUUID(), counterReportTitle, match,
-                      kbTitleName, kbTitleId,
-                      false))).mapEmpty();
-            }
+            UUID kbTitleId = erm != null ? erm.getUUID(0) : null;
+            String kbTitleName = erm != null ? erm.getString(1) : null;
+            Future<Void> future = con.preparedQuery("INSERT INTO " + titleEntriesTable(pool)
```
Comment (julianladisch) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:260]: The  `future` variable is no longer needed.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 3]
```diff
@@ -245,37 +245,27 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
...
+                + "(id, counterReportTitle, matchCriteria,"
+                + " kbTitleName, kbTitleId,"
+                + " kbManualMatch, printISSN, onlineISSN)"
+                + " VALUES ($1, $2, $3, $4, $5, $6, $7, $8)"
+                + " ON CONFLICT (counterReportTitle) DO NOTHING")
+                .execute(Tuple.of(UUID.randomUUID(), counterReportTitle, match,
+                    kbTitleName, kbTitleId,
+                    false, printIssn, onlineIssn)).mapEmpty();
```
Comment (julianladisch) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:268]: mapEmpty() is not needed.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:248]: Using onlineIssn would suffice. The reason for declaring this variable this way is that I think we want to generalize the match key a bit. I hope that's ok.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:260]: ok

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:268]: yep, done

**Comment by adamdickmeiss** [comment]
Thx for review Julian. I think using the `daterange` type is good in this case. If you agree, I'll use that for other date range to follow.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 4]
```diff
@@ -245,43 +245,33 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
...
-            return future.compose(x ->
-                con.preparedQuery("SELECT id FROM " + titleEntriesTable(pool)
-                    + " WHERE counterReportTitle = $1")
-                    .execute(Tuple.of(counterReportTitle))
-                    .map(res2 -> res2.iterator().next().getUUID(0))
-            );
+            UUID kbTitleId = erm != null ? erm.getUUID(0) : null;
+            String kbTitleName = erm != null ? erm.getString(1) : null;
+            return con.preparedQuery("INSERT INTO " + titleEntriesTable(pool)
+                + "(id, counterReportTitle, matchCriteria,"
```
Comment (skoczko) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:261]: @adamdickmeiss In the design doc we have moved the counterReportTitle to the TITLE_DATA table.

Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:261]: Yes. That seems very strange by now.

Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:261]: I will change this in a future PR.

**Pull request merged**